uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature

```diff
@@ -15,12 +15,35 @@ Feature: GGMQ-1
       | aws.greengrass.client.Mqtt5JavaSdkClient | classpath:/greengrass/components/recipes/client_java_sdk.yaml |
     Then the Greengrass deployment is COMPLETED on the device after 300 seconds
     And agent "agent1" is connected to MQTT Client Control
-    When I create client device "test" with the following AWS IoT policy
-      | operation | resource     |
-      | connect   | "*"          |
-      | subscribe | "iot_data_0" |
-      | publish   | "iot_data_0" |
+    When I create client device "test"
     And I associate "test" with ggc
+    And I configure aws.greengrass.clientdevices.Auth component
+    """
+{
+  "deviceGroups": {
+    "formatVersion": "2021-03-05",
+    "definitions": {
+      "MyPermissiveDeviceGroup": {
+        "selectionRule": "thingName: test",
```

Comment:
the thing name should not actually be "test" - this will cause problems when the test is run in parallel on multiple devices. Thing names and anything else which is shared must have uniquely generated names which are replaced at runtime. This is why I suggested `${test}` or similar, as a string replacement.

Reply:
I see. Is it ok, if we change  thing creation step to something like:
"When I create client device 'test' with random prefix in thing name" ?
Name mapping will be stored in the test context.
Then we will be able to use short name in test scenarios and full thing name when it is necessary.
For example:
"""
"selectionRule": "thingName: ${testContext.getThingNameForDevice('test')}",
"""
What do you think?

Reply:
Right, like that. I don't think you need to say "with random prefix in thing name", we can make the assumption that it will need to always be random.

Reply:
Fixed ed88cd4b6ed567a4a912fc9635fa36c4ca271d5b